src/relations/cos.py

```diff
@@ -39,7 +39,7 @@ class COSRelation:
     _METRICS_RELATION_NAME = "metrics-endpoint"
     _LOGGING_RELATION_NAME = "logging"
     _PEER_RELATION_NAME = "cos"
-    _ROUTER_LOG_FILE = "/var/log/mysqlrouter/mysqlrouter.log"
+    _ROUTER_LOG_FILES_TARGET = "/var/log/mysqlrouter/**/*log*"
```

Comment:
I'm not sure this is necessary. Did you observed logs not being captured by promtail? My assumption here is that promtail will be able to capture any lines written to `/var/log/mysqlrouter/mysqlrouter.log` before rotation, but some configuration may be need to logrotate. Check [the promtail and log rotation doc](https://grafana.com/docs/loki/latest/send-data/promtail/logrotation/)

Reply:
The issue is that there may be logs that are rotated prior to the relation with grafana-agent-k8s being formed. Promtail does keep track of fds, so you are right - when `/var/log/mysqlrouter/mysqlrouter.log` is rotated, there will be no duplicate log rows in loki

Reply:
Gotcha